src/components/issues/ActionForm.tsx

```diff
@@ -63,39 +78,47 @@ const ActionForm: FC<ActionFormProps> = ({ taskId }) => {
                 placeholder="Assignee"
                 value={state.assignee}
                 onChange={(e) =>
-                    dispatch({ type: 'assignee', value: e.target.value })
+                    dispatch({
+                        type: 'assignee',
+                        value: e.target.value,
+                    })
                 }
+                disabled={isAssigned}
             />
-            <label htmlFor="ends-on" className={styles.assign_label}>
-                Ends on:
-            </label>
-            <input
-                name="ends-on"
-                id="ends-on"
-                className={styles.assign}
-                type="date"
-                onChange={(e) =>
-                    dispatch({ type: 'endsOn', value: e.target.value })
-                }
-            />
-            <label htmlFor="status" className={styles.assign_label}>
-                Status:
-            </label>
-            <select
-                name="status"
-                id="status"
-                value={state.status}
-                onChange={(e) =>
-                    dispatch({ type: 'status', value: e.target.value })
-                }
-                className={styles.assign}
-            >
-                {taskStatus.map(([name, status]) => (
-                    <option key={status} value={status}>
-                        {beautifyStatus(name)}
-                    </option>
-                ))}
-            </select>
+            {!isAssigned && (
+                <>
+                    <label htmlFor="ends-on" className={styles.assign_label}>
+                        Ends on:
+                    </label>
+                    <input
+                        name="ends-on"
+                        id="ends-on"
+                        className={styles.assign}
+                        type="date"
+                        onChange={(e) =>
+                            dispatch({ type: 'endsOn', value: e.target.value })
+                        }
+                    />
+                    <label htmlFor="status" className={styles.assign_label}>
+                        Status:
+                    </label>
+                    <select
+                        name="status"
+                        id="status"
+                        value={state.status}
+                        onChange={(e) =>
+                            dispatch({ type: 'status', value: e.target.value })
+                        }
+                        className={styles.assign}
+                    >
+                        {taskStatus.map(([name, status]) => (
```

Comment:
are we sure about taskStatus not being undefined or null?

Reply:
yes